### ISSUE TITLE
message_size_limit uninitialized constant Iodine::Http

### ISSUE THREAD

**Issue by moxgeek** [boazsegev/plezi#30]
when i try to put `Iodine::Http::Websockets.message_size_limit = 42000` , i got the excpetion :   `uninitialized constant Iodine::Http`
changing the value on the main file   to  `bundle exec iodine -v -maxms 4194304   -t 16 -w 20 -www ./public` not changing anything ...
my need is to make socket send message for more than 4 Mb .

**Comment by boazsegev** [comment]
I apologize. This is a documentation issue that references an old version of iodine.

I’ll take care of it when I’m by my computer.

For now, please make sure you have the latest version of iodine installed and run `iodine -?` to get a list of command line options.

I recommend that you use these options to adjust any runtime options rather than hard-code these settings to your app.

Kindly,
   Bo.

**Comment by boazsegev** [comment]
Possibly related to [fix in version 0.7.17](https://github.com/boazsegev/iodine/blob/master/CHANGELOG.md#change-log-v0717):

> fixed CLI argument recognition for WebSocket message limits and HTTP header limits. Typos in the CLI argument names prevented the CLI from effecting the default values.

**Comment by boazsegev** [comment]
P.S.

Please notice that the `Iodine::DEFAULT_SETTINGS[:max_msg]` [is in Kb](https://github.com/boazsegev/iodine/blob/ee8d8d663fe9689d02df20ea05aff376d3ec3ecc/ext/iodine/iodine.c#L723-L725) (same as `iodine -maxms ...`).

For 4Mb you need 4096 Kb, which (unless I have a bug somewhere) should translate as:

```bash
bundle exec iodine -v -maxms 4096 -t 16 -w 20 -www ./public
```

Also, this limit is only relevant for **incoming** data. It's a security measure that protects the server against a malicious client sending huge amounts of data as a type of Slowloris WebSocket DoS attack.

Outgoing data is unlimited in size, although I do recommend fragmenting large data (or the client might be overwhelmed).

**Comment by moxgeek** [comment]
> Also, this limit is only relevant for incoming data ... Outgoing data is unlimited in size


 that's weird  , i'm facing this problem with sending data from server to the client ...
the client receives an error when trying to get a bigger data : **failed: Invalid frame header**
with js and java client.

**Comment by boazsegev** [comment]
Hi @moxgeek 

I can't recreate the issue. This is the code I used to send up to 32Mb of data:

```ruby
require 'iodine'

class Client
  def on_open client
    messages = []
    messages << ("a" * 16)
    messages << ("a" * 1024)
    messages << ("a" * (1024 * 1024))
    messages << ("a" * (4 * 1024 * 1024))
    messages << ("a" * (6 * 1024 * 1024))
    messages << ("a" * (32 * 1024 * 1024))
    Iodine.run_every(500, messages.length) { client.write messages.shift; client.close if messages.empty? }
  end
  
  def self.call env
   # check if this is an upgrade request.
   if(env['rack.upgrade?'.freeze] == :websocket)
     env['rack.upgrade'.freeze] = Client.new
     [200, {}, []]
   end
   # simply return the RESPONSE object, no matter what request was received.
   [200, {}, "Ok"]
  end
end

Iodine.listen url: "http://:3000", handler: Client

Iodine.threads = 1
Iodine.start
```

Which version of iodine are you running? Are you sending text of binary data? if text, is the text valid UTF-8 encoded (if not, the browser will complain)?

Kindly,
   Bo

**Comment by boazsegev** [comment]
Perfect, thanks! Now I can reproduce the issue.

As a temporary workaround try:

```ruby
# for text messages
subscribe(CHANNEL+userid.to_s) {|ch, msg| msg.encode!("UTF-8"); write msg }
# # OR for binary messages
# subscribe(CHANNEL+userid.to_s) {|ch, msg| write msg }
```

This will circumvent the pub/sub WebSocket encoder where the bug is likely hiding.

I'll try to fix it in the next few days.

**Comment by boazsegev** [comment]
I think I found and fixed the issue.

Any chance you could install iodine from the GitHub repo and confirm that tissue is solved?

Thanks!
   Bo.

**Comment by moxgeek** [comment]
cheers mate ! working fine .
maybe replacing gem iodine at  Gemfile with 
 `gem 'iodine' , git: 'https://github.com/boazsegev/iodine'` 
will be good to future users .
thanks again .

**Comment by boazsegev** [comment]
Hi @moxgeek ,

Thanks for the update!

I just released the patch as part of the iodine 0.7.18 release.

Thanks for exposing the issue. Your name is now engraved into [facil.io's changelog](https://github.com/boazsegev/facil.io/blob/master/CHANGELOG.md#v-070beta7).

FYI: the GitHub master branch is the development branch. It always has the latest bug fixes... but I also break it sometimes while tinkering.

Cheers!
